packages/x-charts/src/ChartsYAxis/ChartsYAxis.tsx

```diff
@@ -141,25 +141,28 @@ function ChartsYAxis(inProps: ChartsYAxisProps) {
     // - No data is associated to the axis (other scale types)
     return null;
   }
+
   return (
     <AxisRoot
       transform={`translate(${position === 'right' ? left + width : left}, 0)`}
       className={classes.root}
     >
       {!disableLine && (
-        <Line
-          y1={yScale.range()[0]}
-          y2={yScale.range()[1]}
-          className={classes.line}
-          {...slotProps?.axisLine}
-        />
+        <Line y1={top - 1} y2={top + height} className={classes.line} {...slotProps?.axisLine} />
       )}
 
       {yTicks.map(({ formattedValue, offset, labelOffset, value }, index) => {
         const xTickLabel = positionSign * (tickSize + 2);
         const yTickLabel = labelOffset;
         const skipLabel =
           typeof tickLabelInterval === 'function' && !tickLabelInterval?.(value, index);
+
+        const showLabel = offset >= top && offset <= height + top;
```

Comment:
On the X axis I added an extra pixel because on some charts the ticks at the beginning/end of the axis were removed even in initial zoom

```suggestion
        const showLabel = offset >= top-1 && offset <= height + top+1;
```